File: Example/ExampleObjC/Controllers/ChatInTabNavigationController/ChatInTabNavigationController.m
//
//  ChatInTabNavigationController.m
//  ThreadsDemoObjc
//
//  Created by Vitaliy Kuzmenko on 25/07/2019.
//  Copyright © 2019 Threads. All rights reserved.
//

#import "ChatInTabNavigationController.h"
#import <Threads/Threads.h>
#import "AttributesHelper.h"

@implementation ChatInTabNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.viewControllers = @[[self getChatViewController]];
}

- (THRAttributes *)getAttributes {
    switch (self.design) {
        case DesignDefault:
            return [AttributesHelper getDefaultAttributes];
        case DesignAlternative:
            return [AttributesHelper getAltAttributes];
    }
}

- (UIViewController *)getChatViewController {
    THRAttributes *attributes = [self getAttributes];
    UIViewController *chatViewController = [[Threads threads] chatViewControllerWithAttributes:attributes];
    return chatViewController;
}

@end

File: Example/ExampleSwift/Controllers/ChatInTabNavigationController/ChatInTabNavigationController.swift
//
//  ChatInTabNavigationController.swift
//  ExampleSwift
//
//  Created by Vitaliy Kuzmenko on 07/08/2019.
//  Copyright © 2019 Threads. All rights reserved.
//

import UIKit
import Threads

class ChatInTabNavigationController: UINavigationController {

    override func viewDidLoad() {
        super.viewDidLoad()

        self.viewControllers = [self.getChatViewController()]
    }
    
    func getAttributes() -> THRAttributes {
        let attributes = THRAttributes()
        return attributes
    }
    
    func getChatViewController() -> UIViewController {
        let attributes = getAttributes()
        let chatViewController = Threads.threads().chatViewController(with: attributes)
        return chatViewController
    }

}
